data-prepper-plugins/otel-trace-source/src/main/java/org/opensearch/dataprepper/plugins/source/oteltrace/OTelTraceGrpcService.java

```diff
@@ -96,7 +96,7 @@ private void processRequest(
         } catch (Exception e) {
             LOG.error("Failed to parse the request due to:", request, e);
             badRequestsCounter.increment();
-            responseObserver.onError(Status.INVALID_ARGUMENT.withDescription(e.getMessage()).asException());
+            responseObserver.onCompleted();
```

Comment:
This doesn't seem quite right to me. We are recording bad messages as successful. It seems there is a deeper issue here where the threads are being lost on errors. That is probably what we need to find an answer on.

Reply:
As per - https://grpc.github.io/grpc-java/javadoc/io/grpc/stub/StreamObserver.html - "completed" only means, the stream is successfully completed, which is true because we have completed processing. Since we are not doing "onNext", I don't think it is a successful request. That's how I understood the API. What do you mean by "where the threads are being lost"?